lms/djangoapps/course_api/tests/test_api.py

```diff
@@ -0,0 +1,185 @@
+    """
+
+    maxDiff = 5000  # long enough to show mismatched dicts
+
+    expected_course_data = {
+        'course_id': u'edX/toy/2012_Fall',
+        'name': u'Toy Course',
+        'number': u'toy',
+        'org': u'edX',
+        'short_description': u'A course about toys.',
+        'media': {
+            'image': {
+                'uri': u'/c4x/edX/toy/asset/just_a_test.jpg',
+                'name': u'Course Image',
+                'description': u'',
+                'type': u'logo',
+            }
+        },
+        'start': u'2015-07-17T12:00:00Z',
+        'start_type': u'timestamp',
+        'start_display': u'July 17, 2015',
+        'end': u'2015-09-19T18:00:00Z',
+        'enrollment_start': u'2015-06-15T00:00:00Z',
+        'enrollment_end': u'2015-07-15T00:00:00Z',
+        'blocks_url': '/api/courses/v1/blocks/?course_id=edX%2Ftoy%2F2012_Fall',
+    }
+
+    @classmethod
+    def setUpClass(cls):
+        super(CourseApiTestMixin, cls).setUpClass()
+        cls.request_factory = RequestFactory()
+
+
+class TestGetCourseDetail(CourseApiTestMixin, SharedModuleStoreTestCase):
+    """
+    Test course_detail api function
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TestGetCourseDetail, cls).setUpClass()
+        cls.course = cls.create_course()
+        cls.hidden_course = cls.create_course(course=u'hidden', visible_to_staff_only=True)
+        cls.honor_user = cls.create_user('honor', is_staff=False)
+        cls.staff_user = cls.create_user('staff', is_staff=True)
+
+    def _make_api_call(self, requesting_user, target_user, course_key):
+        """
+        Call the `course_detail` api endpoint to get information on the course
+        identified by `course_key`.
+        """
+        request = self.request_factory.get('/')
+        request.user = requesting_user
+        return course_detail(request.user, target_user, course_key, request)
+
+    def test_get_existing_course(self):
+        course_key = CourseKey.from_string(u'edX/toy/2012_Fall')
+        result = self._make_api_call(self.honor_user, self.honor_user.username, course_key)
+        self.assertEqual(self.expected_course_data, result.data)
+
+    def test_get_nonexistent_course(self):
+        course_key = CourseKey.from_string(u'edX/toy/nope')
+        with self.assertRaises(Http404):
+            self._make_api_call(self.honor_user, self.honor_user.username, course_key)
+
+    def test_hidden_course_for_honor(self):
+        course_key = CourseKey.from_string(u'edX/hidden/2012_Fall')
+        with self.assertRaises(Http404):
+            self._make_api_call(self.honor_user, self.honor_user.username, course_key)
+
+    def test_hidden_course_for_staff(self):
+        course_key = CourseKey.from_string(u'edX/hidden/2012_Fall')
+        result = self._make_api_call(self.staff_user, self.staff_user.username, course_key)
+        self.assertIsInstance(result.data, dict)
+
+    def test_hidden_course_for_staff_as_honor(self):
```

Comment:
Args should have been self.staff_user, self.honor_user.username, course_key.  Fixed.